src/App.tsx

```diff
@@ -1,26 +1,299 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos, USER_ID, deleteTodos, addTodo } from './api/todos';
+import { Todo } from './types/Todo';
+import classNames from 'classnames';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('all');
+  const [errorMessage, setErrorMessage] = useState('');
+  const [loadingTodoId, setLoadingTodoId] = useState<number | null>(null);
+  const [newTodoTitle, setNewTodoTitle] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isAdding, setIsAdding] = useState(false);
+
+  const activeCount = todos.filter(todo => todo.completed === false);
+  const completedTodos = todos.filter(todo => todo.completed);
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => setErrorMessage('Unable to load todos'))
+  }, []);
+
+  useEffect(() => {
+    if (!isAdding && inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [isAdding]);
+
+  const filterTodos = () => {
+    if (filter === 'active') {
+      return todos.filter(todo => !todo.completed);
+    } else if (filter === 'completed') {
+      return todos.filter(todo => todo.completed);
```

Comment:
Create a enum for 'all', 'active', 'completed' and use it everywhere. Also, will be better if you use switch case here